src/main/scala/dpla/ingestion3/OaiHarvesterMain.scala

```diff
@@ -2,144 +2,89 @@ package dpla.ingestion3
   */
 object OaiHarvesterMain {
-
-  val recordSchemaStr =
-    """{
-        "namespace": "dpla.avro",
-        "type": "record",
-        "name": "OriginalRecord.v1",
-        "doc": "",
-        "fields": [
-          {"name": "id", "type": "string"},
-          {"name": "document", "type": "string"},
-          {"name": "set_id", "type": "string"},
-          ("name": "set_document", "type": "string"},
-          {"name": "provider", "type": "string"},
-          {"name": "mimetype", "type": { "name": "MimeType",
-           "type": "enum", "symbols": ["application_json", "application_xml", "text_turtle"]}
-           }
-        ]
-      }
-    """//.stripMargin // TODO we need to template the document field so we can record info there
-
-  // This schema String is printed to help with debugging.
-  // It is NOT implemented during the write operation b/c sets are written to CSV.
-  val setSchemaStr =
-    """{
-        "namespace": "dpla.avro",
-        "type": "set",
-        "name": "OriginalRecord.v1",
-        "doc": "",
-        "fields": [
-          {"name": "id", "type": "string"},
-          {"name": "document", "type": "string"},
-          {"name": "provider", "type": "string"},
-          {"name": "mimetype", "type": { "name": "MimeType",
-           "type": "enum", "symbols": ["application_json", "application_xml", "text_turtle"]}
-           }
-        ]
-      }
-    """
-
   val logger = LogManager.getLogger(OaiHarvesterMain.getClass)
 
   def main(args: Array[String]): Unit = {
     val oaiConf = new OaiHarvesterConf(args.toSeq)
-    val oaiParams = oaiConf.load()
 
-    // TODO print something pleasant.
-    Utils.deleteRecursively(new File(oaiParams.outputDir.get))
+    Utils.deleteRecursively(new File(oaiConf.outputDir()))
 
     // Initiate spark session.
     val sparkConf = new SparkConf().setAppName("Oai Harvest")
     // sparkMaster has a default value of local[*] if not provided.
     // TODO: will this default value work with EMR?
-    sparkConf.setMaster(oaiParams.sparkMaster.get)
+    sparkConf.setMaster(oaiConf.sparkMaster())
 
     val spark = SparkSession.builder().config(sparkConf).getOrCreate()
     val sc = spark.sparkContext
+    val schema = new FlatFileIO().readFileAsString("/avro/OriginalRecord.avsc")
+
+    val start = System.currentTimeMillis()
 
-    // Set options
     val readerOptions: Map[String, String] = Map(
-      "verb" -> oaiParams.verb,
-      "metadataPrefix" -> oaiParams.metadataPrefix,
-      "harvestAllSets" -> oaiParams.harvestAllSets,
-      "setlist" -> oaiParams.setList,
-      "blacklist" -> oaiParams.blacklist,
-      "endpoint" -> oaiParams.endpoint
+      "verb" -> oaiConf.verb.toOption,
+      "metadataPrefix" -> oaiConf.prefix.toOption,
+      "harvestAllSets" -> oaiConf.harvestAllSets.toOption,
+      "setlist" -> oaiConf.setlist.toOption,
+      "blacklist" -> oaiConf.blacklist.toOption
     ).collect{ case (key, Some(value)) => key -> value } // remove None values
```

Comment:
Not part of this PR, so ignore, but this line can be replaced with a call to .flatten() or .flatMap() I think